skymp5-server/cpp/lib_espm/espm.cpp

```diff
@@ -566,29 +562,57 @@ const std::vector<espm::RecordHeader*>& espm::Browser::GetRecordsByType(
 }
 
 const std::vector<espm::RecordHeader*>& espm::Browser::GetRecordsAtPos(
-  uint32_t cellOrWorld, int16_t cellX, int16_t cellY)
+  uint32_t cellOrWorld, int16_t cellX, int16_t cellY) const
 {
   return pImpl->cellOrWorldChildren[RefrKey(cellOrWorld, cellX, cellY)];
 }
 
-bool espm::Browser::ReadAny(void* parentGrStack)
+namespace espm {
+
+const GroupStack* Browser::GetParentGroupsOptional(const RecordHeader* rec) const {
+  return pImpl->GetParentGroupsOptional(rec);
+}
+
+const GroupStack& Browser::GetParentGroupsEnsured(const RecordHeader* rec) const {
+  const auto opt = GetParentGroupsOptional(rec);
+  if (!opt) {
+    throw std::runtime_error("espm::Browser: no parent groups for record");
+  }
+  return *opt;
+}
+
+const std::vector<void*>* Browser::GetSubsOptional(const GroupHeader* group) const {
+  return pImpl->GetSubsOptional(group);
+}
+
+const std::vector<void*>& Browser::GetSubsEnsured(const GroupHeader* group) const {
+  const auto opt = GetSubsOptional(group);
+  if (!opt) {
+    throw std::runtime_error("espm::Browser: no subs for record");
+  }
+  return *opt;
+}
+
+}
+
+bool espm::Browser::ReadAny(const GroupStack* parentGrStack)
 {
-  if (pImpl->pos >= pImpl->length)
+  if (pImpl->pos >= pImpl->length) {
     return false;
+  }
 
-  char* pType = pImpl->buf + pImpl->pos;
+  std::string pType(pImpl->buf + pImpl->pos, 4);
```

Comment:
Might be significantly slower at least in Debug on MSVC implementation. There are alternatives such as `espm::Type`. Or we could leave it as it is

Reply:
Yeah, no reason to change this now, I just forgot to clean some extra stuff I experimented with

Reply:
Reverted this